packages/gatsby/src/redux/reducers/jobsv2.ts

```diff
@@ -1,10 +1,20 @@
-module.exports = (
-  state = { incomplete: new Map(), complete: new Map() },
-  action
-) => {
+import { ActionsUnion, IGatsbyState, IGatsbyJobV2 } from "../types"
+
+const jobDefaultValue = {
+  contentDigest: ``,
+  inputPaths: [],
+}
+
+export const jobsV2Reducer = (
+  state: IGatsbyState["jobsV2"] = {
+    incomplete: new Map(),
+    complete: new Map(),
+  },
+  action: ActionsUnion
+): IGatsbyState["jobsV2"] => {
   switch (action.type) {
     case `CREATE_JOB_V2`: {
-      const { job, plugin } = action.payload
+      const { job = jobDefaultValue, plugin } = action.payload
```

Comment:
We don't want to create empty jobs. Better to noop than add an empty job to the queue.

Reply:
So, just to confirm: here in the reducer should I return the default state if job is undefined:
```js
if (!job) return state
```

As same as the `remove-stale-job`?